packages/start/src/client/createServerFn.ts

```diff
@@ -216,21 +217,49 @@ export function createServerFn<
           ...extractedFn,
           // The extracted function on the server-side calls
           // this function
-          __executeServer: (opts: any) =>
-            executeMiddleware(resolvedMiddleware, 'server', {
+          __executeServer: (opts: any) => {
+            const parsedOpts =
+              opts instanceof FormData ? extractFormDataContext(opts) : opts
+
+            return executeMiddleware(resolvedMiddleware, 'server', {
               ...extractedFn,
-              ...opts,
+              ...parsedOpts,
             }).then((d) => ({
               // Only send the result and sendContext back to the client
               result: d.result,
               context: d.sendContext,
-            })),
+            }))
+          },
         },
       ) as any
     },
   }
 }
 
+function extractFormDataContext(formData: FormData) {
+  const serializedContext = formData.get('__TSR_CONTEXT')
+  formData.delete('__TSR_CONTEXT')
+
+  if (typeof serializedContext !== 'string') {
+    return {
+      context: {},
+      data: formData,
+    }
+  }
+
+  try {
+    const context = defaultTransformer.parse(serializedContext)
+    return {
+      context,
+      data: formData,
+    }
+  } catch (e) {
+    return {
+      data: formData,
+    }
+  }
```

Comment:
I need others' opinions on this piece since it allows the request to be processed if the context cannot be parsed. However, I'm not sure if this is the desired behaviour, and we'd rather throw an error on context deserialisation failure.